src/Control/Email/Mailer.php

```diff
@@ -15,513 +13,124 @@ class Mailer extends Object
-                "<BODY bgColor=\"#ffffff\">\n" .
-                    $htmlContent .
-                "\n</BODY>\n" .
-                "</HTML>";
+        if (!$this->transport) {
+            $transportClass = $this->config()->swift_transport;
+            $this->setTransport(new $transportClass());
         }
-
-        // Make the HTML part
-        $headers["Content-Type"] = "text/html; charset=utf-8";
-        $headers["Content-Transfer-Encoding"] = $this->getMessageEncoding();
-        $htmlContentEncoded = $this->encodeMessage($htmlContent, $this->getMessageEncoding());
-
-        // Merge with headers
-        return $this->processHeaders($headers, $htmlContentEncoded);
+        return $this->transport;
     }
 
     /**
-     * Encode an array of parts using multipart
-     *
-     * @param array $parts List of parts
-     * @param string $contentType Content-type of parts
-     * @param array $headers Existing headers to include in response
-     * @return array Array with two items, the body followed by headers
+     * @param \Swift_Transport $transport
+     * @return $this
      */
-    protected function encodeMultipart($parts, $contentType, $headers = array())
+    public function setTransport($transport)
     {
-        $separator = "----=_NextPart_" . preg_replace('/[^0-9]/', '', rand() * 10000000000);
-
-        $headers["MIME-Version"] = "1.0";
-        $headers["Content-Type"] = "$contentType; boundary=\"$separator\"";
-        $headers["Content-Transfer-Encoding"] = "7bit";
-
-        if ($contentType == "multipart/alternative") {
-            // $baseMessage = "This is an encoded HTML message.  There are two parts: a plain text and an HTML message,
-            // open whatever suits you better.";
-            $baseMessage = "\nThis is a multi-part message in MIME format.";
-        } else {
-            // $baseMessage = "This is a message containing attachments.  The e-mail body is contained in the first
-            // attachment";
-            $baseMessage = "\nThis is a multi-part message in MIME format.";
-        }
-
-        $separator = "\n--$separator\n";
-        $body = "$baseMessage\n" .
-            $separator . implode("\n".$separator, $parts) . "\n" . trim($separator) . "--";
-
-        return array($body, $headers);
+        $this->transport = $transport;
+        return $this;
     }
 
-
     /**
-     * Add headers to the start of the message
-     *
-     * @param array $headers
-     * @param string $body
-     * @return string Resulting message body
+     * @return \Swift_Mailer
      */
-    protected function processHeaders($headers, $body = '')
+    public function getSwiftMailer()
     {
-        $result = '';
-        foreach ($headers as $key => $value) {
-            $result .= "$key: $value\n";
-        }
-        if ($body) {
-            $result .= "\n$body";
+        if (!$this->swift) {
+            $this->setSwiftMailer(new \Swift_Mailer($this->getTransport()));
```

Comment:
We could inject this with dependencies, instead of via PHP.

```
Injector:
  Swift_Transport: Swift_MailTransport
  Swift_Mailer:
    constructor:
      - '%$Swift_Transport'
  SilverStripe\Control\Email\Mailer:
    properties:
      SwiftMailer: '%$Swift_Mailer'
```

How about this?